src/IssueViz/Editor/LocationTagging/LocationTaggerProvider.cs

```diff
@@ -61,10 +61,16 @@ internal class LocationTaggerProvider : ITaggerProvider
                 return null;
             }
 
-            // Need to make sure we only create one per buffer
-            var tagger = buffer.Properties.GetOrCreateSingletonProperty(typeof(LocationTagger),
-                () => new LocationTagger(buffer, locationStore, spanCalculator, logger));
+            // We only want one location tagger per buffer and we don't want it be disposed until
+            // it is not being used by any tag aggregators, so we're wrapping it in a SingletonDisposableTaggerManager
+            var taggerManager = buffer.Properties.GetOrCreateSingletonProperty(typeof(SingletonDisposableTaggerManager<IIssueLocationTag>),
+                () => new SingletonDisposableTaggerManager<IIssueLocationTag>(InternalCreateTagger));
+
+            var tagger = taggerManager.CreateTagger(buffer);
```

Comment:
This is the only change to the existing product code - switching the location tagger provider to create the singleton manager rather than the tagger directly.